class.json-api-endpoints.php

```diff
@@ -267,11 +267,28 @@ function input( $return_default_values = true, $cast_and_filter = true ) {
 			break;
 		}
 
-		if ( !$cast_and_filter ) {
-			return $return;
+		if ( isset( $this->api->query['force'] ) && 'secure' === $this->api->query['force'] ) {
+			if ( isset( $return['secure_key'] ) ) {
+				$post_body = $this->get_secure_body( $return['secure_key'] );
+				$this->api->post_body = $post_body;
+				$this->api->query['force'] = false;
+				return $this->input( $return_default_values, $cast_and_filter );
+			}
+		}
+
+		if ( $cast_and_filter ) {
+			$return = $this->cast_and_filter( $return, $this->request_format, $return_default_values );
 		}
+		return $return;
+	}
 
-		return $this->cast_and_filter( $return, $this->request_format, $return_default_values );
+
+	protected function get_secure_body( $secure_key ) {
+		$response =  Jetpack_Client::wpcom_json_api_request_as_blog( sprintf( '/sites/%d/secure-request', Jetpack_Options::get_option('id' ) ), '1.1', array( 'method' => 'POST' ), array( 'secure_key' => $secure_key ) );
```

Comment:
This line is exceptionally long, consider spreading it over several lines, e.g.

$response =  Jetpack_Client::wpcom_json_api_request_as_blog( 
sprintf( '/sites/%d/secure-request', Jetpack_Options::get_option('id' ) ), 
'1.1', 
array( 'method' => 'POST' ), 
array( 'secure_key' => $secure_key )
 );